### ISSUE TITLE
Unable to locate executable file: Unity.exe

### ISSUE THREAD

**Issue by CodusMaximus** [Dinomite-Studios/unity-azure-pipelines-tasks#158]
I've recently installed Unity build into my DevOps organisation, created 2 pipelines (Windows & Android).  I followed both tutorials floating about with regards to creating an Azure DevOps Pipeline however I get this error every time:

Unable to locate executable file: 'C:\Program Files\Unity\Hub\Editor\2019.4.14f1\Editor\Unity.exe'. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also verify the file has a valid extension for an executable file.

However, the file exists on my computer:
![image](https://user-images.githubusercontent.com/9824539/103485635-b1c60800-4def-11eb-9fe0-6cc961636e53.png)

Apologies if this is a DevOps Self-Hosted Agent issue, I don't know where else to turn

**Comment by pawciut** [comment]
I have the same problem. Also tried adding agents permission to this folder but still get the error.

**Comment by CodusMaximus** [comment]
Hey man, have you just downloaded/installed the latest version from the marketplace just now?

Can you confirm a running build?

**Comment by CodusMaximus** [comment]
> I have the same problem. Also tried adding agents permission to this folder but still get the error.
> 
> EDIT:
> Ok, i found the error in my configuration
> 
> When I had "Editors folder location" set as "C:\Program Files\2020.1.17f1\Editor"
> Pipeline logged an error that it cannot find: "C:\Program Files\2020.1.17f1\Editor\2020.1.17f1\Editor\Unity.exe"
> 
> So I've changed value of editors path to just: "C:\Program Files" and it worked

Hey man, have you just downloaded/installed the latest version from the marketplace just now?

Can you confirm a running build?

**Comment by pawciut** [comment]
> > I have the same problem. Also tried adding agents permission to this folder but still get the error.
> > EDIT:
> > Ok, i found the error in my configuration
> > When I had "Editors folder location" set as "C:\Program Files\2020.1.17f1\Editor"
> > Pipeline logged an error that it cannot find: "C:\Program Files\2020.1.17f1\Editor\2020.1.17f1\Editor\Unity.exe"
> > So I've changed value of editors path to just: "C:\Program Files" and it worked
> 
> Hey man, have you just downloaded/installed the latest version from the marketplace just now?
> 
> Can you confirm a running build?

Yes, UnityBuildTask finished successfuly after 8m (running agent on my local machine)

![unity ci](https://user-images.githubusercontent.com/16131463/105615715-f3384a80-5dd2-11eb-923c-ba6f215a29d9.PNG)

**Comment by CodusMaximus** [comment]
Cool! that's awesome!

Thanks so much for the reply! Please help just a little more..
 - So is your Unity installed where the blue console text says it is?
 - When you installed the build agent with PowerShell did you give your local account or use networkservice?

Thanks again,

**Comment by pawciut** [comment]
Yes, the blue line is the correct path for unity editor but in configuration i specified only Program Files and the task somehow gets the rest of the path by itself

This was my configuration of the agent
![unity ci agent](https://user-images.githubusercontent.com/16131463/105618765-4918ec80-5deb-11eb-84f6-785b0391036a.PNG)

**Comment by FejZa** [comment]
That is correct. The task does not want to know where an exact version of the unity Editor is but where the folder is, that contains ALL unity installations in the agent.

So e.g. when using Unity Hub, the hub will install all unity editors to Program Files/Unity Hub/Editor/

And then inside that folder the subfolders will be e.g.

- 2019.4.1f1
- 2020.1.1f1
- 2020.2.1f1
...

The task will lookup the required editor version for the project and then add the last bit to the path automatically. Does that make sense?

Let me know if you have more questions.

**Comment by pawciut** [comment]
Yes it make sense, especially after your deeper explanation.
Now that I look at the tooltip it also make sense. But at first glance I was convinced that specific path would be for the folder that contains Unity.exe, also the error said specifically that it cannot file Unity.exe so I think I've got worked up on this thought.

I don't know if there could be different tooltip or caption for this property. Writing this I also noticed there is "Unity editor**s** location". Maybe I was just tired :sweat_smile:

Anyway, thanks for the explanation!
![unity ci tooltip](https://user-images.githubusercontent.com/16131463/105683350-d5aed200-5ef3-11eb-9868-dd2d21611441.png)
